File: languages/php/Bloock/Entity/IdentityV2/DecimalEnumAttributeDescriptor.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\DecimalEnumAttributeDefinitionV2;
use Google\Protobuf\Internal\RepeatedField;

/**
 * Represents a descriptor for an attribute with a decimal enum value.
 */
class DecimalEnumAttributeDescriptor extends AttributeDescriptor
{
    private RepeatedField $enumeration;

    /**
     * Constructs an DecimalEnumAttributeDescriptor object with the specified parameters.
     * @param string $displayName
     * @param string $technicalName
     * @param string|null $description
     * @param bool $required
     * @param RepeatedField $enumeration
     */
    public function __construct(string $displayName, string $technicalName, ?string $description, bool $required, RepeatedField $enumeration)
    {
        parent::__construct($displayName, $technicalName, $description, $required);
        $this->enumeration = $enumeration;
    }

    public static function fromProto(DecimalEnumAttributeDefinitionV2 $res): DecimalEnumAttributeDescriptor
    {
        return new DecimalEnumAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired(), $res->getEnum());
    }

    public function toProto(): DecimalEnumAttributeDefinitionV2
    {
        $p = new DecimalEnumAttributeDefinitionV2();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        $p->setEnum($this->enumeration);
        return $p;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/DecimalEnumAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.IdentityEntities;
import java.util.List;

/**
 * Represents a descriptor for an attribute with a decimal enum value.
 */
public class DecimalEnumAttributeDescriptor extends AttributeDescriptor {
  private List<Double> enumeration;

  /**
   * Constructs an DecimalEnumAttributeDescriptor object with the specified parameters.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   * @param enumeration
   */
  public DecimalEnumAttributeDescriptor(
      String displayName,
      String technicalName,
      String description,
      Boolean required,
      List<Double> enumeration) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public static DecimalEnumAttributeDescriptor fromProto(
      IdentityEntities.DecimalEnumAttributeDefinition res) {
    return new DecimalEnumAttributeDescriptor(
        res.getDisplayName(),
        res.getId(),
        res.getDescription(),
        res.getRequired(),
        res.getEnumList());
  }

  public IdentityEntities.DecimalEnumAttributeDefinition toProto() {
    return IdentityEntities.DecimalEnumAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .addAllEnum(this.enumeration)
        .build();
  }
}
